src/app/test/test.component.ts

```diff
@@ -205,15 +205,17 @@ export class TestComponent implements OnInit {
     });
   }
 
-  openDeleteModal(): void {
+  openDeleteModal(deleteAllReports: boolean): void {
     const reportsToBeDeleted: TestListItem[] = this.getSelectedReports();
-    if (reportsToBeDeleted.length > 0) {
-      this.deleteModal.open(reportsToBeDeleted);
+    if (this.reports && (reportsToBeDeleted.length > 0 || (deleteAllReports && this.reports.length > 0))) {
+      this.deleteModal.open(reportsToBeDeleted, deleteAllReports);
     }
   }
```

Comment:
Maybe trigger a warning toast that there are no reports selected from the `else`?

Reply:
I added the toast